### PR TITLE
WT-4986 Migrate wiredtiger-linux-directio

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -1264,6 +1264,22 @@ tasks:
             set -o verbose
             sh dist/s_clang-scan 2>&1
 
+  - name: directio
```
Comment (lukech) [inline test/evergreen.yml]: Prefer retaining `linux-` in the task name.

File: test/evergreen.yml [hunk 2]
```diff
@@ -1264,6 +1264,22 @@ tasks:
...
+    depends_on:
+    - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "compile wiredtiger"
```
Comment (lukech) [inline test/evergreen.yml]: Jenkins job has the configure options `../configure --enable-diagnostic --enable-strict`. Not sure if other options (as part of the standard `compile wiredtiger` func will make any difference for the test output. Adding @ddanderson for comment.

File: test/evergreen.yml [hunk 3]
```diff
@@ -1264,6 +1264,22 @@ tasks:
...
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/build_posix/test/format"
+          script: |
+            set -o errexit
+            set -o verbose
+            for i in {1..3}; do
+              ./t -1 -C "direct_io=[data]" -c ../../../test/format/CONFIG.stress || cleanup
```
Comment (lukech) [inline test/evergreen.yml]: `|| cleanup` is a legacy for Jenkins, which is not needed here.

File: test/evergreen.yml (hunk 2 above)
Comment (quchenhao) [reply test/evergreen.yml]: I think other things are just extras.

**Pull request merged**